modules/cloudflarepurger/src/Plugin/Purge/Purger/CloudFlarePurger.php

```diff
@@ -0,0 +1,210 @@
+ * CloudFlare purger.
+ *
+ * @PurgePurger(
+ *   id = "cloudflare",
+ *   label = @Translation("CloudFlare"),
+ *   description = @Translation("Purger for CloudFlare."),
+ *   configform = "Drupal\cloudflare\Form\CloudFlareAdminSettingsForm",
+ *   types = {"tag", "url", "everything"},
+ *   multi_instance = FALSE,
+ * )
+ */
+class CloudFlarePurger extends PurgerBase implements PurgerInterface {
+
+  /**
+   * The settings configuration.
+   *
+   * @var \Drupal\Core\Config\Config
+   */
+  protected $config;
+
+  /**
+   * A logger instance.
+   *
+   * @var \Psr\Log\LoggerInterface
+   */
+  protected $logger;
+
+  /**
+   * Tracks rate limits associated with CloudFlare Api.
+   *
+   * @var \Drupal\cloudflare\CloudFlareStateInterface
+   */
+  protected $state;
+
+  /**
+   * ZoneApi object for interfacing with CloudFlare Php Sdk.
+   *
+   * @var \CloudFlarePhpSdk\ApiEndpoints\ZoneApi
+   */
+  protected $zoneApi;
+
+  /**
+   * The current cloudflare ZoneId.
+   *
+   * @var string
+   */
+  protected $zone;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('config.factory'),
+      $container->get('cloudflare.state'),
+      $container->get('logger.factory')->get('cloudflare')
+    );
+  }
+
+  /**
+   * Constructs a \Drupal\Component\Plugin\CloudFlareTagPurger.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config
+   *   The factory for configuration objects.
+   * @param \Drupal\cloudflare\CloudFlareStateInterface $state
+   *   Tracks rate limits associated with CloudFlare Api.
+   * @param \Psr\Log\LoggerInterface $logger
+   *   A logger instance.
+   *
+   * @throws \LogicException
+   *   Thrown if $configuration['id'] is missing, see Purger\Service::createId.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, ConfigFactoryInterface $config, CloudFlareStateInterface $state, LoggerInterface $logger) {
+    $this->config = $config->get('cloudflare.settings');
+    $this->state = $state;
+    $this->logger = $logger;
+
+    // This is a unique case where the ApiSdk is being accessed directly and not
+    // via a service.  Purging should only ever happen through the purge module
+    // which is why this is NOT in a service.
+    $api_key = $this->config->get('apikey');
+    $email = $this->config->get('email');
+    $this->zone = $this->config->get('zone');
+    $this->zoneApi = new ZoneApi($api_key, $email);
+
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
```

Comment:
Mainly a style nitpick, but I would put the parent call at the top of the method body, first things first.